pkg/nodediscovery/nodediscovery.go

```diff
@@ -440,15 +439,13 @@ func (n *NodeDiscovery) mutateNodeResource(nodeResource *ciliumv2.CiliumNode) er
 		APIVersion: "v1",
 		Kind:       "Node",
 		Name:       nodeTypes.GetName(),
-		UID:        k8sNode.UID,
+		UID:        slimNode.UID,
 	}}
-	providerID = k8sNode.Spec.ProviderID
+	providerID = slimNode.Spec.ProviderID
 
 	// Get the addresses from k8s node and add them as part of Cilium Node.
 	// Cilium Node should contain all addresses from k8s.
-	nodeInterface := k8s.ConvertToNode(k8sNode)
```

Comment:
This seems to be the last use of `k8s.ConvertToNode` function outside of test code. We should probably remove it (doesn't have to be done in this PR though)

Reply:
@gandro thanks for catching this. I created https://github.com/cilium/cilium/issues/25680 to track the removal of the func.